iCubBlinker compilation failure against YARP-3.6

Since https://github.com/robotology/yarp/pull/2780 has been merged, `funny-things` is failing with this error:
~~~
2021-12-22T16:04:29.7621615Z -- Build files have been written to: /__w/robotology-superbuild/robotology-superbuild/build/src/funny-things
2021-12-22T16:04:37.1402008Z [235/248] Performing build step for 'funny-things'
2021-12-22T16:04:37.1403659Z FAILED: src/funny-things/CMakeFiles/YCMStamp/funny-things-build /__w/robotology-superbuild/robotology-superbuild/build/src/funny-things/CMakeFiles/YCMStamp/funny-things-build 
2021-12-22T16:04:37.1406115Z cd /__w/robotology-superbuild/robotology-superbuild/build/src/funny-things && /usr/bin/cmake --build . && /usr/bin/cmake -E touch /__w/robotology-superbuild/robotology-superbuild/build/src/funny-things/CMakeFiles/YCMStamp/funny-things-build
2021-12-22T16:04:37.1407593Z [1/10] Generating code from iCubBlinker.thrift
2021-12-22T16:04:37.1408531Z [2/10] Building CXX object modules/iCubBreather/CMakeFiles/iCubBreather.dir/iCubBreatherModule.cpp.o
2021-12-22T16:04:37.1409786Z [3/10] Building CXX object modules/iCubBreather/CMakeFiles/iCubBreather.dir/iCubBreatherThread.cpp.o
2021-12-22T16:04:37.1410864Z [4/10] Building CXX object modules/iCubBlinker/CMakeFiles/iCubBlinker.dir/main.cpp.o
2021-12-22T16:04:37.1411610Z [5/10] Linking CXX executable bin/iCubBreather
2021-12-22T16:04:37.1412181Z [6/10] Running utility command for copy_lua_in_build
2021-12-22T16:04:37.1412965Z [7/10] Building CXX object modules/iCubBlinker/CMakeFiles/iCubBlinker.dir/iCubBlinker_IDL.cpp.o
2021-12-22T16:04:37.1413921Z FAILED: modules/iCubBlinker/CMakeFiles/iCubBlinker.dir/iCubBlinker_IDL.cpp.o 
2021-12-22T16:04:37.1417364Z /usr/bin/c++  -I/__w/robotology-superbuild/robotology-superbuild/build/src/funny-things/modules/iCubBlinker -isystem /__w/robotology-superbuild/robotology-superbuild/build/install/include -g -std=c++1z -MD -MT modules/iCubBlinker/CMakeFiles/iCubBlinker.dir/iCubBlinker_IDL.cpp.o -MF modules/iCubBlinker/CMakeFiles/iCubBlinker.dir/iCubBlinker_IDL.cpp.o.d -o modules/iCubBlinker/CMakeFiles/iCubBlinker.dir/iCubBlinker_IDL.cpp.o -c /__w/robotology-superbuild/robotology-superbuild/build/src/funny-things/modules/iCubBlinker/iCubBlinker_IDL.cpp
2021-12-22T16:04:37.1420920Z /__w/robotology-superbuild/robotology-superbuild/build/src/funny-things/modules/iCubBlinker/iCubBlinker_IDL.cpp:17:11: error: "'blink_fast' will never be called, since 'blink' starts with the same tag"
2021-12-22T16:04:37.1422404Z  YARP_COMPILER_ERROR("'blink_fast' will never be called, since 'blink' starts with the same tag")
2021-12-22T16:04:37.1423012Z            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2021-12-22T16:04:37.1424418Z /__w/robotology-superbuild/robotology-superbuild/build/src/funny-things/modules/iCubBlinker/iCubBlinker_IDL.cpp:18:11: error: "'blink_naturalistic' will never be called, since 'blink' starts with the same tag"
2021-12-22T16:04:37.1425976Z  YARP_COMPILER_ERROR("'blink_naturalistic' will never be called, since 'blink' starts with the same tag")
2021-12-22T16:04:37.1426627Z            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2021-12-22T16:04:37.1427333Z [8/10] Building CXX object modules/funnyPostures/CMakeFiles/funnyPostures.dir/main.cpp.o
2021-12-22T16:04:37.1428109Z ninja: build stopped: subcommand failed.
~~~

fyi @randaz81 @S-Dafarra @pattacini @Nicogene @drdanz

--- a/app/scripts/shells/ADP_demo.sh
+++ b/app/scripts/shells/ADP_demo.sh
@@ -50,7 +50,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 
@@ -266,12 +266,12 @@ noi_robot(){
   sleep 1.0
   me_and_you
   sleep 3.0
-  go_home  
+  go_home
 }
 
 questo_film(){
   show_iit
-  speak "In questo breve filmato vi mostrerò come le mie capacità si siano evolute nel corso degli anni." 
+  speak "In questo breve filmato vi mostrerò come le mie capacità si siano evolute nel corso degli anni."
   sleep 3.0
   mostra_muscoli
   sleep 1.0

--- a/app/scripts/shells/D_G_drawing.sh
+++ b/app/scripts/shells/D_G_drawing.sh
@@ -63,7 +63,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 
@@ -150,7 +150,7 @@ draw() {
     else if [ "$HAND" = "left" ]; then
         echo "$1 $2 $(bc <<< "$3+$OFFSET") $4 $5 $6 $7" | yarp write ...  /armCtrl/left_arm/xd:i
     fi
-    fi  
+    fi
 }
 
 start_draw() {
@@ -210,7 +210,7 @@ right_arm_home(){
 
 set_up_left_arm(){
     echo "ctpq time 1.5 off 7 pos (23.3295 30.7783 -3.06519 53.8881 49.0925 58.8869 45.9779 83.2985 140.312)" | yarp rpc /ctpservice/left_arm/rpc
-    
+
     #with the glove
     #echo "ctpq time 1.5 off 8 pos (9.80532 0 139.999 47.7522 132.133 76.9429 109.127 270.001)" | yarp rpc /ctpservice/left_arm/rpc
 }
@@ -246,7 +246,7 @@ set_up_hand() {
         echo "Using left hand"
         set_up_left_arm
     fi
-    fi    
+    fi
 }
 
 prepare() {
@@ -289,7 +289,7 @@ calib_table()
 }
 
 part_1() {
-    
+
     if [ "$HAND" = "right" ]; then
         #high left arm drawing
         echo "ctpq time 10 off 0 pos (-15.0 0.0 -11.0 -10.0 -0.0 5.0)" | yarp rpc /ctpservice/head/rpc
@@ -311,7 +311,7 @@ part_1() {
 
         #up face face drawing
         draw -0.386334435421571142921 0.0753170618625696063342 -0.040025084774475305771 -0.0558197697426994418612 0.889354215875829745563 -0.453798669025990086823 2.68235527666180484374
-    
+
     else if [ "$HAND" = "left" ]; then
         #high left arm drawing
         draw -0.312099692973232367699 -0.0323477147963253594543 -0.0629028330966882788799 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
@@ -340,7 +340,7 @@ start_part_2() {
     X=-0.325723724067253908032
     Y=0.061830599363296217863
     Z=-0.0253764683221865031681
-    AX=-0.0846392633486617862459 
+    AX=-0.0846392633486617862459
     AY=0.850400373120677621763
     AZ=-0.519283545373823929303
     THETA=2.80263529378769327138
@@ -376,7 +376,7 @@ part_2() {
 
         draw -0.210324055274786971825 0.0726375747705533834075 -0.0732430696781040821985 -0.331076573016411412897 0.839284378792964691485 -0.431265619211427697621 2.33134272817010890222
         sleep 2.0
-    
+
         draw -0.210324055274786971825 0.0726375747705533834075 -0.0192430696781040821985 -0.331076573016411412897 0.839284378792964691485 -0.431265619211427697621 2.33134272817010890222
     else if [ "$HAND" = "left" ]; then
         draw -0.325723724067253908032 -0.061830599363296217863 -0.0753764683221865031681 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
@@ -391,7 +391,7 @@ part_2() {
 
         draw -0.210324055274786971825 -0.0726375747705533834075 -0.0732430696781040821985 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
         sleep 2.0
-    
+
         draw -0.210324055274786971825 -0.0726375747705533834075 -0.0192430696781040821985 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
     fi
     fi
@@ -412,7 +412,7 @@ start_part_3() {
         AZ=$AZ_LEFT
         THETA=$THETA_LEFT
     fi
-    
+
     start_draw $X $Y $Z $AX $AY $AZ $THETA
     sleep 1.0
 
@@ -433,13 +433,13 @@ part_3() {
     if [ "$HAND" = "right" ]; then
         draw -0.315723724067253908032 0.11 -0.0733764683221865031681 -0.0846392633486617862459 0.850400373120677621763 -0.519283545373823929303 2.80263529378769327138
         sleep 2.0
-     
+
         draw -0.2823724067253908032 0.10 -0.0733764683221865031681 -0.0846392633486617862459 0.850400373120677621763 -0.519283545373823929303 2.80263529378769327138
-        sleep 2.0 
-    
+        sleep 2.0
+
         draw -0.2653724067253908032 0.115 -0.0713764683221865031681 -0.0846392633486617862459 0.850400373120677621763 -0.519283545373823929303 2.80263529378769327138
         sleep 2.0
-         
+
         draw -0.2023250129427876709 0.118728067975726200523 -0.0796862721934695412696 -0.00496676038412719334653 0.774492962781589078247 -0.632563026024350216758 2.56487352093593168334
         sleep 3.0
 
@@ -447,13 +447,13 @@ part_3() {
     else if [ "$HAND" = "left" ]; then
         draw -0.315723724067253908032 -0.11 -0.0733764683221865031681 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
         sleep 2.0
-     
+
         draw -0.2823724067253908032 -0.10 -0.0733764683221865031681 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
-        sleep 2.0 
-    
+        sleep 2.0
+
         draw -0.2653724067253908032 -0.115 -0.0713764683221865031681 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
         sleep 2.0
-         
+
         draw -0.2023250129427876709 -0.118728067975726200523 -0.0796862721934695412696 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
         sleep 3.0
 
@@ -490,7 +490,7 @@ part_4() {
     if [ "$HAND" = "right" ]; then
         draw -0.350578938749672666564 0.0803265228440144745115 -0.069040502935731338785 -0.139080462476080801704 0.864773202310446520436 -0.482518324546513943663 2.60992081676853482364
         sleep 2.0
-   
+
         draw -0.355263690567933779096 0.110372969207294786109 -0.0714483792852473856283 0.000476032722984540655753 0.899948998521016729768 -0.435994923656073951612 2.64223014371933695443
         sleep 2.0
 
@@ -501,7 +501,7 @@ part_4() {
     else if [ "$HAND" = "left" ]; then
         draw -0.350578938749672666564 -0.0803265228440144745115 -0.069040502935731338785 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
         sleep 2.0
-   
+
         draw -0.355263690567933779096 -0.110372969207294786109 -0.0714483792852473856283 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
         sleep 2.0
 
@@ -528,10 +528,10 @@ start_part_5() {
         AZ=$AZ_LEFT
         THETA=$THETA_LEFT
     fi
-    
+
     start_draw $X $Y $Z $AX $AY $AZ $THETA
     sleep 1.0
-    
+
     echo "ctpq time 2 off 0 pos (-25.0 0.0 -11.0 -20.0 -0.0 5.0)" | yarp rpc /ctpservice/head/rpc
 
     torso_low
@@ -539,7 +539,7 @@ start_part_5() {
 }
 
 part_5() {
-    
+
      echo "ctpq time 5 off 0 pos (-25.0 0.0 -18.0 -20.0 -0.0 5.0)" | yarp rpc /ctpservice/head/rpc
 
     if [ "$HAND" = "right" ]; then
@@ -547,7 +547,7 @@ part_5() {
         sleep 2.0
 
         draw -0.306886776896212300603 0.127579528816639274136 -0.0788054075826516292613 -0.0406193229074667538914 0.812088367509156250357 -0.582119020444146206827 2.56000409079834900794
-        sleep 2.0 
+        sleep 2.0
 
         draw -0.306886776896212300603 0.127579528816639274136 -0.0488054075826516292613 -0.0406193229074667538914 0.812088367509156250357 -0.582119020444146206827 2.56000409079834900794
     else if [ "$HAND" = "left" ]; then
@@ -555,7 +555,7 @@ part_5() {
         sleep 2.0
 
         draw -0.306886776896212300603 -0.127579528816639274136 -0.0788054075826516292613 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
-        sleep 2.0 
+        sleep 2.0
 
         draw -0.306886776896212300603 -0.127579528816639274136 -0.0488054075826516292613 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
     fi
@@ -601,7 +601,7 @@ part_6() {
         draw -0.288886776896212300603 -0.129579528816639274136 -0.0788054075826516292613 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
         sleep 2.0
 
-        draw -0.288886776896212300603 -0.129579528816639274136 -0.0488054075826516292613 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT  
+        draw -0.288886776896212300603 -0.129579528816639274136 -0.0488054075826516292613 $AX_LEFT $AY_LEFT $AZ_LEFT $THETA_LEFT
     fi
     fi
 }
@@ -609,7 +609,7 @@ part_6() {
 raise_up() {
 
     #draw -0.288886776896212300603 0.129579528816639274136 -0.0488054075826516292613 -0.0406193229074667538914 0.812088367509156250357 -0.582119020444146206827 2.56000409079834900794
-  
+
     echo $HAND
     X=-0.288886776896212300603
     Y=0.205579528816639274136
@@ -627,7 +627,7 @@ raise_up() {
     fi
     echo "ctpq time 3 off 0 pos (-25.0 0.0 -5.0 -18.0 -0.0 5.0)" | yarp rpc /ctpservice/head/rpc
 
-    draw $X $Y $Z $AX $AY $AZ $THETA 
+    draw $X $Y $Z $AX $AY $AZ $THETA
 }
 
 all_parts() {
@@ -648,7 +648,7 @@ all_parts() {
     sleep 2.0
     part_3
     sleep 2.0
-    
+
     start_part_4
     sleep 2.0
     part_4
@@ -664,50 +664,50 @@ all_parts() {
     part_6
     sleep 2.0
 
-    raise_up    
+    raise_up
 }
 
 ###################################################################
-# 21 feb 
+# 21 feb
 ###################################################################
 
-speak_01() 
+speak_01()
 {
     echo "\"Ciao Domenico e Stefano!!!\"" | yarp write ... /iSpeak
-    mplayer /home/vvasco/stilisti/audio_01.mp3 
+    mplayer /home/vvasco/stilisti/audio_01.mp3
 }
 
-#speak_02() 
+#speak_02()
 #{
 #    echo "\"Come sono felice ed emozionato di essere qui a Milano tra voi!!!!\"" | yarp write ... /iSpeak
 #    mplayer /home/vvasco/stilisti/audio_02.mp3
 #}
 
-#speak_03() 
+#speak_03()
 #{
 #    echo "\"Cosa mi potete insegnare????\"" | yarp write ... /iSpeak
-#    mplayer /home/vvasco/stilisti/audio_03.mp3 
+#    mplayer /home/vvasco/stilisti/audio_03.mp3
 #}
 
-speak_02() 
+speak_02()
 {
     echo "\"Vorrei tanto imparare a disegnare!!!!\"" | yarp write ... /iSpeak
     mplayer /home/vvasco/stilisti/audio_02.mp3
 }
 
-#speak_05() 
+#speak_05()
 #{
 #    echo "\"Come posso aiutarvi per preparare questa sfilata???\"" | yarp write ... /iSpeak
 #    mplayer /home/vvasco/stilisti/audio_05.mp3
 #}
 
-#speak_06() 
+#speak_06()
 #{
 #    echo "\"Wow che bel look!!!\"" | yarp write ... /iSpeak
 #    mplayer /home/vvasco/stilisti/audio_06.mp3
 #}
 
-#speak_07() 
+#speak_07()
 #{
 #    echo "\"Grazie Stefano e Domenico, ci vediamo alla sfilata!!!\"" | yarp write ... /iSpeak
 #    mplayer /home/vvasco/stilisti/audio_07.mp3
@@ -721,7 +721,7 @@ speak_02()
 
 
 ###################################################################
-# 22 feb 
+# 22 feb
 ###################################################################
 speak_03()
 {
@@ -729,32 +729,32 @@ speak_03()
     mplayer /home/vvasco/stilisti/audio_03.mp3
 }
 
-speak_04() 
+speak_04()
 {
     echo "\"Wow che bel look!!!\"" | yarp write ... /iSpeak
     mplayer /home/vvasco/stilisti/audio_04.mp3
 }
 
-speak_05() 
+speak_05()
 {
     echo "\"Wow che bello!!!\"" | yarp write ... /iSpeak
     mplayer /home/vvasco/stilisti/audio_05.mp3
 }
 
 
-speak_06() 
+speak_06()
 {
     echo "\"Interessante questo articolo!!!\"" | yarp write ... /iSpeak
     mplayer /home/vvasco/stilisti/audio_06.mp3
 }
 
-speak_07() 
+speak_07()
 {
     echo "\"Grazie Stefano e Domenico!!!\"" | yarp write ... /iSpeak
     mplayer /home/vvasco/stilisti/audio_07.mp3
 }
 
-speak_08() 
+speak_08()
 {
     echo "\"Grazie Stefano e Domenico, ci vediamo alla sfilata!!!\"" | yarp write ... /iSpeak
     mplayer /home/vvasco/stilisti/audio_08.mp3
@@ -811,16 +811,16 @@ point_left_front() {
 }
 
 follow_left() {
-    point_left_wide 
-    point_left_front 
+    point_left_wide
+    point_left_front
     sleep 5.0
-    home_left_arm 
+    home_left_arm
 }
 
 follow_both() {
     for i in {1..$1}
     do
-        follow_left 
+        follow_left
         sleep 6.0
         follow_right
         sleep 6.0
@@ -1023,7 +1023,7 @@ sequence_14() {
     sleep 0.7
     echo "ctpq time 1.0 off 0 pos (-22.0 34.0 48.0 73.0 37.0 3.0 -7.0    12.0 -6.0 37.0 2.0 0.0 3.0 2.0 1.0 0.0)" | yarp rpc /ctpservice/left_arm/rpc
     sleep 3.0
-   
+
     home_left_arm
     home_right_arm
 }

--- a/app/scripts/shells/baricub.sh
+++ b/app/scripts/shells/baricub.sh
@@ -21,7 +21,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 

--- a/app/scripts/shells/blinking-experiments.sh
+++ b/app/scripts/shells/blinking-experiments.sh
@@ -25,7 +25,7 @@ usage() {
 cat << EOF
 ***************************************************************************************
 BLINKING EXPERIMENTS
-Author:  Alessandro Roncone   <alessandro.roncone@iit.it> 
+Author:  Alessandro Roncone   <alessandro.roncone@iit.it>
 
 This script scripts through the commands available for the DeA Kids videos.
 
@@ -51,7 +51,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
 }
 
 wait_till_quiet() {
@@ -70,7 +70,7 @@ speak_hri() {
     speak "$1"
     sleep 0.15
     wait_till_quiet
-    # blink    
+    # blink
 }
 
 #do a blink only at the beginning (onset)
@@ -134,7 +134,7 @@ echo ""
 
 $1 "$2"
 
-if [[ $# -eq 0 ]] ; then 
+if [[ $# -eq 0 ]] ; then
     echo "[BlinkingScript] No options were passed!"
     echo ""
     usage

--- a/app/scripts/shells/corriereInnovazione.sh
+++ b/app/scripts/shells/corriereInnovazione.sh
@@ -54,7 +54,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 
@@ -464,7 +464,7 @@ seq_01() {
 
 seq_02() {
     speak "Mi chiamo aicab!"
-    speak "Piacere di conoscervi!" 
+    speak "Piacere di conoscervi!"
     sleep 1.0 && blink && hello_both
     wait_till_quiet
     go_home_walking
@@ -525,7 +525,7 @@ seq_10() {
 
 #### Day 01/06
 seq2_01() {
-    speak "Buongiorno a tutti!" && hello_both 
+    speak "Buongiorno a tutti!" && hello_both
     go_home_walking
     speak "Io sono aicab, il robot dalle sembianze di un bambino."
     sleep 1.0 && blink

--- a/app/scripts/shells/dea-scripting.sh
+++ b/app/scripts/shells/dea-scripting.sh
@@ -25,7 +25,7 @@ usage() {
 cat << EOF
 ***************************************************************************************
 DEA SCRIPTING
-Author:  Alessandro Roncone   <alessandro.roncone@iit.it> 
+Author:  Alessandro Roncone   <alessandro.roncone@iit.it>
 
 This script scripts through the commands available for the DeA Kids videos.
 
@@ -50,7 +50,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 
@@ -310,7 +310,7 @@ closing_remarks() {
     sleep 3.0 && blink && smile
     speak "In bocca al lupo meikers"
     smile
-    greet_with_right_thumb_up   
+    greet_with_right_thumb_up
     smile
 }
 
@@ -369,7 +369,7 @@ angry() {
         sleep 1.0 && blink
         sleep 1.0 && smile
         speak "E' stata usata una stampante 3D? proprio come quelle del laboratorio degli X meikers!"
-        sleep 4.0 && blink 
+        sleep 4.0 && blink
         speak " Solo? Un po' piu' grande."
         sleep 6.0 && smile # && blink
     }
@@ -650,7 +650,7 @@ angry() {
         sleep 3.0 && blink
         sleep 2.0 && smile
     }
-    
+
     puntata1_3() {
         attacco_grafica "Il 70% dei ragazzi ha rilevato? nei normali monopattini, un grosso difetto: utilizzarli a lungo e' molto faticoso."
         sleep 3.0 && blink
@@ -712,7 +712,7 @@ angry() {
         sleep 2.0 && blink
         sleep 2.0 && blink
     }
-    
+
     puntata2_5() {
         breathers "stop"
         echo "ctpq time 1.5 off 0 pos (-42.0 36.0 -12.0 101.0 85.0 -45.0 -4.0 17.0 57.0 87.0 140.0 0.0 0.0 87.0 176.0 250.0)" | yarp rpc /ctpservice/right_arm/rpc
@@ -732,7 +732,7 @@ angry() {
     }
 
     puntata2_c() {
-        closing_remarks "0 3 FT2"  
+        closing_remarks "0 3 FT2"
     }
 
     puntata2_total() {
@@ -761,7 +761,7 @@ angry() {
         meteo_bot
         speak "Le macchinine si bloccano quando affrontano alcuni tipi di terreno, come la ghiaia, il fango? o i tappeti. Il 67% dei ragazzi lo trova insopportabile."
         sleep 4.0 && blink
-        sleep 2.0 && blink 
+        sleep 2.0 && blink
         sleep 6.0 && blink
     }
 
@@ -788,7 +788,7 @@ angry() {
         sleep 1.0 && blink
         sleep 1.0 && blink
         sleep 3.0 && blink
-    }    
+    }
 
     puntata3_c() {
         closing_remarks "3 6 TL 2 4"
@@ -818,7 +818,7 @@ angry() {
         meteo_bot
         speak "Il 50 percento dei ragazzi vorrebbe un televisore meno anonimo, piu' colorato? piu' giocoso? piu' adatto allo stile delle loro camerette."
         sleep 4.0 && blink
-        sleep 4.0 && blink 
+        sleep 4.0 && blink
     }
 
     puntata4_3() {
@@ -838,10 +838,10 @@ angry() {
         attacco_grafica "Infine, il 60 percento dei ragazzi perde il telecomando almeno una volta al giorno. La percentuale sale al 90 percento tra i ragazzi particolarmente disordinati che lasciano vestiti e giocattoli in giro."
         sleep 3.0 && blink
         sleep 6.0 && blink
-    }  
+    }
 
     puntata4_c() {
-        closing_remarks "0 5 ics, 1 2"  
+        closing_remarks "0 5 ics, 1 2"
     }
 
 #######################################################################################
@@ -860,7 +860,7 @@ angry() {
         meteo_bot
         speak "Il 70 percento degli umani vorrebbe un taabl-et in grado di distinguersi dalla massa. Dovrebbe farsi notare anche quando non lo si usa."
         sleep 3.0 && blink
-        sleep 7.0 && blink 
+        sleep 7.0 && blink
     }
 
     puntata5_3() {
@@ -876,7 +876,7 @@ angry() {
         attacco_grafica "Infine? il 79% vorrebbe poter condividere quello che c'e' sullo schermo con i propri amici, anche quando gli amici sono tanti e sono tutti nello stesso posto."
         sleep 1.0 && blink
         sleep 3.0 && blink
-    }    
+    }
 
     puntata5_c() {
         closing_remarks "1 9 t 6 2"
@@ -899,7 +899,7 @@ angry() {
         meteo_bot
         speak "Il 70 percento dei ragazzi vorrebbe una bicicletta aggressiva, simile a una moto. I ragazzi la definiscono tosta? anche se aicab non capisce il significato di questa parola."
         sleep 5.0 && blink
-        sleep 4.0 && no_testa 
+        sleep 4.0 && no_testa
         echo "set reb cun" | yarp rpc /icub/face/emotions/in
         echo "set leb cun" | yarp rpc /icub/face/emotions/in
         sleep 3.0 && smile
@@ -918,7 +918,7 @@ angry() {
     }
 
     puntata6_c() {
-        closing_remarks "0 4 y 5 4"  
+        closing_remarks "0 4 y 5 4"
     }
 
 #######################################################################################
@@ -953,7 +953,7 @@ angry() {
     puntata7_5() {
         attacco_grafica "Per finire? il 65 percento sogna uno zaino da supereroe in grado di proteggerli dagli scherzi dei compagni, e soprattutto, dai bulli della scuola."
         sleep 2.0  && blink
-        sleep 3.0  && angry 
+        sleep 3.0  && angry
         sleep 2.25 && angry
         sleep 2.0  && smile
     }
@@ -1158,7 +1158,7 @@ angry() {
         sleep 0.5
         # speak "Rimarra' a disposizione di tutti gli aspiranti meikers? nella galleria dei prototipi? sul sito vuvuvu punto dea kidz punto it"
         speak "Rimarra' a disposizione di tutti gli aspiranti meikers? nella galleria dei prototipi sul nostro sito."
-        
+
         head "start"
         sleep 16.0
         breathers "stop"
@@ -1169,7 +1169,7 @@ angry() {
         sleep 1.5
         breathers "start"
     }
-    
+
     chiusura_1() {
         chiusura "Il protocollo? Super Monopattino e' chiuso e archiviato."
     }
@@ -1232,7 +1232,7 @@ echo ""
 
 $1 "$2"
 
-if [[ $# -eq 0 ]] ; then 
+if [[ $# -eq 0 ]] ; then
     echo "No options were passed!"
     echo ""
     usage

--- a/app/scripts/shells/enel.sh
+++ b/app/scripts/shells/enel.sh
@@ -54,7 +54,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 
@@ -339,9 +339,9 @@ sequence_02() {
     sleep 1.0
     show_musles
     speak "Sono nato grazie alle idee di tanti super mega ingegneri!. Ecco perche' qui all'ENEL mi sento a casa... A proposito: buon pomeriggio presidente, ben trovato Ing. Starace e ciao a tutti ragazzi!"
-    sleep 2.0    
-    greet_with_right_thumb_up    
-    
+    sleep 2.0
+    greet_with_right_thumb_up
+
     hello_both
     wait_till_quiet
     smile && blink
@@ -360,7 +360,7 @@ sequence_03() {
     sleep 1.0 && blink
     speak "Famoso come Robo kop. E Jig robot. Voglio stampare gli autografi in 3D. e farmi i selfie con le star di Hollywood."
     speak "E quindi, per cominciare, ho pensato di salire su questo palco e provare a fare il presentatore"
-    
+
     sleep 15.0 && blink
     fonzie
     speak "Dai mettimi alla prova, Fammi fare un'intervista"
@@ -377,7 +377,7 @@ sequence_04() {
     wait_till_quiet
     smile
     sleep 2.0
-    # look presidente 
+    # look presidente
     gaze "look 15.0 0.0 3.0"
     speak "Presidente, quale' secondo lei, la caratteristica che i bambini di oggi, crescendo, non dovranno mai dimenticare, per lavorare in un contesto fortemente automatizzato"
     smile && blink
@@ -386,7 +386,7 @@ sequence_04() {
 
 
 sequence_05() {
-    # look Ingegnere 
+    # look Ingegnere
     gaze "look 15.0 0.0 3.0"
     sleep 2.0
     speak "Ingegnere, quali sono secondo lei, le conoscenze di cui avra' bisogno Enel, per svolgere i lavori del futuro?"
@@ -403,7 +403,7 @@ sequence_06() {
     #speak "Dott. Starace, ma se io sono figlio di ingegneri ... e lei e' un ingegnere, noi siamo anche un po parenti?"
     #wait_till_quiet
     speak "ora vi devo lasciare. ma torno presto, e' troppo bello stare sul palco di wi are enerji! CIAO CIAO!"
-    sleep 4.0    
+    sleep 4.0
     hello_right
     wait_till_quiet
     smile && blink
@@ -413,7 +413,7 @@ sequence_06() {
 sequence_07() {
     gaze "look 15.0 0.0 3.0"
     speak "Eccomi! sono qui, scusami... ma mi sono dovuto allontanare un momento."
-    sleep 1.0    
+    sleep 1.0
     hello_right_simple
     smile && blink
 }

--- a/app/scripts/shells/interaction-scripting.sh
+++ b/app/scripts/shells/interaction-scripting.sh
@@ -114,7 +114,7 @@ attend_left() {
 #######################################################################################
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 
@@ -123,16 +123,16 @@ blink() {
 #######################################################################################
 
 point_right() {
-    echo "point (-0.430766 0.204718 -0.127849) right no_head" | yarp rpc /actionsRenderingEngine/cmd:io &    
-    sleep 4.0     
+    echo "point (-0.430766 0.204718 -0.127849) right no_head" | yarp rpc /actionsRenderingEngine/cmd:io &
+    sleep 4.0
     home_gaze
     sleep 2.0
     gaze_person
 }
 
 point_left() {
-    echo "point (-0.430766 -0.204718 -0.127849) left no_head" | yarp rpc /actionsRenderingEngine/cmd:io &    
-    sleep 4.0     
+    echo "point (-0.430766 -0.204718 -0.127849) left no_head" | yarp rpc /actionsRenderingEngine/cmd:io &
+    sleep 4.0
     home_gaze
     sleep 2.0
     gaze_person
@@ -145,15 +145,15 @@ expect_right() {
 
 expect_left() {
     look_hand_left
-    echo "expect left near" | yarp rpc /actionsRenderingEngine/cmd:io   
+    echo "expect left near" | yarp rpc /actionsRenderingEngine/cmd:io
 }
 
 look_hand_right() {
-    echo "look hand right" | yarp rpc /actionsRenderingEngine/cmd:io   
+    echo "look hand right" | yarp rpc /actionsRenderingEngine/cmd:io
 }
 
 look_hand_left() {
-    echo "look hand left" | yarp rpc /actionsRenderingEngine/cmd:io 
+    echo "look hand left" | yarp rpc /actionsRenderingEngine/cmd:io
 }
 
 return_home() {

--- a/app/scripts/shells/mezzorainpiu.sh
+++ b/app/scripts/shells/mezzorainpiu.sh
@@ -36,7 +36,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 
@@ -358,7 +358,7 @@ sequence_02() {
     go_home_helper 2.0
     wait_till_quiet
     smile && blink
-   
+
 }
 
 sequence_03() {
@@ -366,7 +366,7 @@ sequence_03() {
     sleep 1.0
     speak "No, preferisco rimanere un robot, e affiancare gli umani nella vita di tutti i giorni, cercando di svolgere al meglio le mansioni che mi verranno affidate."
     sleep 1.0
-    fonzie_right 
+    fonzie_right
     sleep 2.0
     go_home_helper 2.0
     sleep 2.0 && blink
@@ -380,11 +380,11 @@ sequence_04() {
     gaze "look-around 15.0 0.0 5.0"
     sleep 1.0
     speak "Perchè, i robot, e le tecnologie sviluppate grazie a me, potranno aiutarvi nella cura e nel supporto alla disabilità, a casa come pure al lavoro"
-    sleep 2.0    
-    fonzie_right    
+    sleep 2.0
+    fonzie_right
     speak "farvi correre meno rischi e risparmiare le risorse del nostro pianeta, che inizia ad essere un po' sovraffollato."
     sleep 10.0
-    gaze "look -15.0 0.0 5.0"    
+    gaze "look -15.0 0.0 5.0"
     show_iit
     gaze "look-around 15.0 0.0 5.0"
     wait_till_quiet

--- a/app/scripts/shells/pd-scripting.sh
+++ b/app/scripts/shells/pd-scripting.sh
@@ -54,7 +54,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 
@@ -367,12 +367,12 @@ sequence_00() {
 
 sequence_01() {
 #    gaze "idle"
-    look_presentatore 
+    look_presentatore
     speak "Ciao Fabrizio."
     sleep 1.0
     look_pubblico
     speak "Sono contento di essere qui con voi oggi."
-    sleep 1.0 
+    sleep 1.0
 #   gaze "idle"
     speak "Perche non ci divertiamo un po'?"
     wait_till_quiet
@@ -488,7 +488,7 @@ sequence_09()
     smile
 }
 
-sequence_10(){ 
+sequence_10(){
 look_schermo
 speak "Grazie ragazzi!"
 sleep 1.0
@@ -500,7 +500,7 @@ smile
 }
 
 
-sequence_11(){ 
+sequence_11(){
 look_presentatore
 torso_home
 speak "Grazie, anche per me Ee stato divertente!"

--- a/app/scripts/shells/president.sh
+++ b/app/scripts/shells/president.sh
@@ -54,7 +54,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 

--- a/app/scripts/shells/rai-scripting.sh
+++ b/app/scripts/shells/rai-scripting.sh
@@ -25,7 +25,7 @@ usage() {
 cat << EOF
 ***************************************************************************************
 DEA SCRIPTING
-Author:  Alessandro Roncone   <alessandro.roncone@iit.it> 
+Author:  Alessandro Roncone   <alessandro.roncone@iit.it>
 
 This script scripts through the commands available for the DeA Kids videos.
 
@@ -50,7 +50,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 
@@ -172,7 +172,7 @@ wait_till_quiet() {
     isSpeaking=$(echo "stat" | yarp rpc /iSpeak/rpc)
     while [ "$isSpeaking" == "Response: speaking" ]; do
         isSpeaking=$(echo "stat" | yarp rpc /iSpeak/rpc)
-        sleep 0.1        
+        sleep 0.1
         # echo $isSpeaking
     done
     echo "I'm not speaking any more :)"
@@ -183,7 +183,7 @@ wait_till_quiet() {
 # FUNZIONI LANCIO
 #######################################################################################
 
-vittoria() {    
+vittoria() {
     echo "ctpq time 1.0 off 7 pos                                       (18.0 40.0 50.0 167.0 0.0 0.0 0.0 0.0 222.0)" | yarp rpc /ctpservice/$1/rpc
     echo "ctpq time 2.0 off 0 pos (-57.0 32.0 -1.0 88.0 56.0 -30.0 -11.0 18.0 40.0 50.0 167.0 0.0 0.0 0.0 0.0 222.0)" | yarp rpc /ctpservice/$1/rpc
 }
@@ -195,7 +195,7 @@ lancio_destro() {
     sleep 3.0 && smile && sleep 3.0
 
     go_home_helperR 2.0
-    breathers "start"    
+    breathers "start"
 }
 
 
@@ -206,7 +206,7 @@ lancio_sinistro() {
     sleep 3.0 && smile && sleep 3.0
 
     go_home_helperL 2.0
-    breathers "start"  
+    breathers "start"
 }
 
 lancio_due_mani() {
@@ -217,7 +217,7 @@ lancio_due_mani() {
     sleep 3.0 && smile && sleep 3.0
 
     go_home
-    breathers "start"  
+    breathers "start"
 }
 
 #######################################################################################
@@ -226,7 +226,7 @@ lancio_due_mani() {
     #risposta_1() {
     #    breathers "start"
     #    smile
-    #    sleep 0.5 
+    #    sleep 0.5
     #    speak "Grazie Annalisa per avermi invitato."
     #    sleep 1.0 && blink
     #    sleep 2.5 && blink
@@ -236,7 +236,7 @@ lancio_due_mani() {
     risposta_1() {
         breathers "start"
         smile
-        sleep 0.5 
+        sleep 0.5
         speak "Grazie Annalisa per avermi invitato."
         sleep 1.0 && blink
         wait_till_quiet
@@ -249,16 +249,16 @@ lancio_due_mani() {
         sleep 1.0 && blink
         wait_till_quiet
         blink
-        smile      
+        smile
     }
 
-    risposta_3() {        
-        sleep 0.5 
+    risposta_3() {
+        sleep 0.5
         speak "cucciolo intelligente, perche' ho le dimensioni e la forma di un bambino, e come i bambini posso imparare cose nuove."
         sleep 3.0 && blink
         wait_till_quiet
         blink
-        smile   
+        smile
     }
 
     risposta_4() {
@@ -266,16 +266,16 @@ lancio_due_mani() {
         sleep 2.0 && blink
         wait_till_quiet
         blink
-        smile   
+        smile
     }
 
     risposta_5() {
-        speak "sono nato dieci anni fa, dal disegno degli ingegneri che mi hanno costruito? e in questi anni mi hanno insegnato a compiere azioni come un bambino di quattro anni" 
+        speak "sono nato dieci anni fa, dal disegno degli ingegneri che mi hanno costruito? e in questi anni mi hanno insegnato a compiere azioni come un bambino di quattro anni"
         sleep 2.0 && blink
         sleep 1.0 && blink
         wait_till_quiet
         blink
-        smile   
+        smile
     }
 
     risposta_6() {
@@ -283,18 +283,18 @@ lancio_due_mani() {
         sleep 2.0 && blink
         wait_till_quiet
         blink
-        smile 
-        sleep 0.4  
+        smile
+        sleep 0.4
 
         speak "Sono capace di stare in piedi senza cadere, e ora sto imparando a camminare come voi umani."
         sleep 1.5 && blink
         wait_till_quiet
         blink
-        smile   
+        smile
     }
 
     risposta_7() {
-        sleep 0.5 
+        sleep 0.5
         speak "la mia casa e' all'Istituto Italiano di Tecnologia a Genova, ma prima di venire a RaiNext ero a Ma drid, per incontrare scienziati interessati a sviluppare la mia intelligenza."
         sleep 2.0 && blink
         sleep 2.5 && blink
@@ -305,10 +305,10 @@ lancio_due_mani() {
         sleep 3.0 && blink
         wait_till_quiet
         blink
-        smile   
+        smile
     }
-    
-    
+
+
     risposta_8() {
         speak "oltre a me, esistono altri 30 aicab nel mondo, in laboratori che studiano l'intelligenza artificiale, come negli Stati Uniti? in Giappone e in alcune citta' Europee."
         sleep 2.0 && blink
@@ -318,7 +318,7 @@ lancio_due_mani() {
     }
 
     risposta_9() {
-        
+
         speak "Mi hanno inventato scienziati italiani ed europei per realizzare una macchina in grado di sviluppare un'intelligenza simile a quella degli umani. Il responsabile del mio progetto e' Giorgio Metta? all'Istituto Italiano di Tecnologia, che e' uno dei miei papa'. ma ho anche tanti zii e amici che lavorano per migliorarmi.. Tra qualche anno? potro' aiutarvi nei lavori domestici o in altre attivita' delicate per cui avrete bisogno di supporto. come per esempio l'assistenza degli anziani\!"
         sleep 5.0 && blink
         sleep 5.0 && blink
@@ -326,17 +326,17 @@ lancio_due_mani() {
         sleep 5.0 && blink
         wait_till_quiet
         blink
-        smile   
+        smile
     }
 
     risposta_10() {
 
-        
+
         speak "Sono un robot particolare\! un robot umanoide\! cioe' con le sembianze simili a un essere umano."
         sleep 5.0 && blink
         wait_till_quiet
         blink
-        smile   
+        smile
     }
 
     risposta_11() {
@@ -344,39 +344,39 @@ lancio_due_mani() {
         sleep 4.0 && blink
         wait_till_quiet
         blink
-        smile   
+        smile
     }
 
     risposta_12() {
 
-        
+
         blink && sleep 0.5
         speak "Sono fatto di metallo, plastica? e circuiti elettrici. ma i miei progettisti stanno pensando a nuovi materiali, per rendermi piu' leggero e sicuro"
         sleep 5.0 && blink
         wait_till_quiet
         blink
-        smile   
+        smile
     }
 
     risposta_13() {
 
-        
+
         speak "in questo momento? costo duecentocinquantamila euro, ma il mio progetto e' oo pen surs. Su internet si trova il disegno per costruire il mio corpo e il softuer per la base della mia intelligenza\!"
         sleep 8.0 blink
         wait_till_quiet
         blink
         blink
-        smile   
+        smile
     }
 
     risposta_14() {
 
-        
+
         speak "In futuro costero' meno di una siticar, sotto i diecimila euro. cosi' che le persone potranno comprarmi per avermi in casa? o in un ambiente lavorativo, e aiutarli in diversi compiti."
         sleep 8.0 && blink
         wait_till_quiet
         blink
-        smile   
+        smile
     }
 
     risposta_15() {
@@ -385,7 +385,7 @@ lancio_due_mani() {
         sleep 9.0 && blink
         sleep 5.0 && blink && blink
         wait_till_quiet
-        smile   
+        smile
     }
 
 #######################################################################################
@@ -396,7 +396,7 @@ echo ""
 
 $1 "$2"
 
-if [[ $# -eq 0 ]] ; then 
+if [[ $# -eq 0 ]] ; then
     echo "No options were passed!"
     echo ""
     usage

--- a/app/scripts/shells/rete4.sh
+++ b/app/scripts/shells/rete4.sh
@@ -54,7 +54,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 
@@ -343,8 +343,8 @@ show_agitation() {
     echo "bind roll 0.0 0.0" | yarp rpc /iKinGazeCtrl/rpc
     sleep 0.2
     echo "bind yaw 0.0 0.0" | yarp rpc /iKinGazeCtrl/rpc
-    sleep 0.2    
-    
+    sleep 0.2
+
     gaze "look -30.0 0.0 5.0"
     sleep 1.5
     gaze "look 30.0 0.0 5.0"
@@ -360,7 +360,7 @@ show_agitation() {
     sleep 0.2
     echo "clear yaw" | yarp rpc /iKinGazeCtrl/rpc
     sleep 0.2
-    
+
     echo "ctpq time 1.0 off 0 pos (0.0 0.0 0.0)" | yarp rpc /ctpservice/torso/rpc
 }
 
@@ -408,7 +408,7 @@ sequence_01() {
 sequence_02() {
     speak "Ho solo 10 anni, e per me e' tardi, dovrei andare a dormire, tra poco. ma stasera staro' sveglio. Questa occasione non posso perderla!"
     sleep 1.0
-    greet_with_right_thumb_up    
+    greet_with_right_thumb_up
     wait_till_quiet
     smile && blink
 }
@@ -422,26 +422,26 @@ sequence_03_() {
     sequence_04
 }
 
-sequence_04() {    
+sequence_04() {
     speak "Si, sono creature bellissime, e ci sono tanti cuccioli come me! Ma, morderanno??"
     sleep 1.5
     #greet_with_right_thumb_up
 }
 
-sequence_05() {  
-    surprised  
+sequence_05() {
+    surprised
     sleep 1.0
     speak "Questi sono davvero tanti!...Vai avanti tu, ACP, che devi presentarli. io ti seguo...da qua."
     smile
 }
 
-sequence_06() {   
-    victory right_arm 
+sequence_06() {
+    victory right_arm
     speak "Sono decisamente piu' carino, e molto piu' giovane!."
     go_home
 }
 
-sequence_07() {   
+sequence_07() {
     speak "Belli i felini! Saranno bravi genitori? Guardiamo, io vi aspetto dopo."
     sleep 1.0
     question
@@ -461,7 +461,7 @@ sequence_09() {
 
 sequence_10() {
     speak "Oltre 30, nati dopo di me, e sparsi per il mondo, dagli stati uniti al giappone."
-    sleep 3.5    
+    sleep 3.5
     question_right
     sleep 1.5
     question_left
@@ -472,7 +472,7 @@ sequence_10() {
 
 sequence_11() {
     speak "Ho tante mamme e papa'; all'Istituto Italiano di Tecnologia di Genova. Oggi mi ha accompagnato qui Vadiim "
-    sleep 2.0    
+    sleep 2.0
     hello_left_simple
     go_home_helper 2.0
 }
@@ -499,18 +499,18 @@ sequence_13() {
     #go_home
 }
 
-sequence_14() { 
+sequence_14() {
     speak "Io restero' con i ricercatori, e continuero' a imparare e crescere. sara' R1 ad entrare nelle vostre case per aiutarvi"
 }
 
 # Skin DEMO.
 
-sequence_15() { 
+sequence_15() {
     speak "16961, ti dice niente questo numero?"
     interaction_right
 }
 
-sequence_16() { 
+sequence_16() {
     speak "Grazie Alessandro, e' ora che io vada a dormire. Mi e' piaciuta la TV."
     echo "ctpq time 1.0 off 0 pos (-12.0 37.0 6.0 67.0 -52.0 -14.0 9.0    12.0 -6.0 37.0 2.0 0.0 3.0 2.0 1.0 0.0)" | yarp rpc /ctpservice/left_arm/rpc
     sleep 2.0

--- a/app/scripts/shells/tg2.sh
+++ b/app/scripts/shells/tg2.sh
@@ -54,7 +54,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 

--- a/app/scripts/shells/tim-scripting.sh
+++ b/app/scripts/shells/tim-scripting.sh
@@ -16,7 +16,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 
@@ -25,19 +25,19 @@ look_at() {
 }
 
 take_pen() {
-    echo "ctpq time 3.0 off 7 pos (27.0 47.0 22.0 123.0 66.0 100.0 35.0 90.0 240.0)" | yarp rpc /ctpservice/right_arm/rpc  
+    echo "ctpq time 3.0 off 7 pos (27.0 47.0 22.0 123.0 66.0 100.0 35.0 90.0 240.0)" | yarp rpc /ctpservice/right_arm/rpc
 }
 
 home() {
     look_at -0.9 0.0 0.34
     echo "ctpq time 1.0 off 0 pos (0.0 0.0 0.0)" | yarp rpc /ctpservice/torso/rpc
-    echo "ctpq time 1.5 off 0 pos (0.0 34.0 4.0 69.0 9.0 0.0 -1.0 54.0 30.0 9.0 49.0 33.0 49.0 30.0 26.0 73.0)" | yarp rpc /ctpservice/left_arm/rpc  
-    echo "ctpq time 1.5 off 0 pos (-6.0 42.0 17.0 83.0 -10.0 0.0 26.0 27.0 47.0 22.0 123.0 66.0 100.0 35.0 90.0 240.0)" | yarp rpc /ctpservice/right_arm/rpc  
+    echo "ctpq time 1.5 off 0 pos (0.0 34.0 4.0 69.0 9.0 0.0 -1.0 54.0 30.0 9.0 49.0 33.0 49.0 30.0 26.0 73.0)" | yarp rpc /ctpservice/left_arm/rpc
+    echo "ctpq time 1.5 off 0 pos (-6.0 42.0 17.0 83.0 -10.0 0.0 26.0 27.0 47.0 22.0 123.0 66.0 100.0 35.0 90.0 240.0)" | yarp rpc /ctpservice/right_arm/rpc
     sleep 2.0
 }
 
 retract_left() {
-    echo "ctpq time 0.8 off 0 pos (2.0 45.0 20.0 104.0 -3.0 0.0 -1.0 54.0 30.0 9.0 49.0 33.0 49.0 30.0 26.0 73.0)" | yarp rpc /ctpservice/left_arm/rpc  
+    echo "ctpq time 0.8 off 0 pos (2.0 45.0 20.0 104.0 -3.0 0.0 -1.0 54.0 30.0 9.0 49.0 33.0 49.0 30.0 26.0 73.0)" | yarp rpc /ctpservice/left_arm/rpc
 }
 
 draw_roof() {
@@ -66,7 +66,7 @@ point() {
     look_at -0.5 0.14 -0.05
     sleep 1.0
     echo "ctpq time 1.0 off 0 pos (12.0 0.0 0.0)" | yarp rpc /ctpservice/torso/rpc
-    echo "ctpq time 1.0 off 0 pos (-6.0 22.0 33.0 66.0 75.0 0.0 6.0 54.0 30.0 46.0 100.0 0.0 0.0 60.0 130.0 220.0)" | yarp rpc /ctpservice/left_arm/rpc  
+    echo "ctpq time 1.0 off 0 pos (-6.0 22.0 33.0 66.0 75.0 0.0 6.0 54.0 30.0 46.0 100.0 0.0 0.0 60.0 130.0 220.0)" | yarp rpc /ctpservice/left_arm/rpc
     sleep 2.0
     look_at -0.5 0.0 0.34
     sleep 2.0
@@ -77,14 +77,14 @@ look_you_old() {
     echo "set mou neu" | yarp rpc /icub/face/emotions/in
     echo "set reb neu" | yarp rpc /icub/face/emotions/in
     echo "set leb neu" | yarp rpc /icub/face/emotions/in
-    echo "ctpq time 0.2 off 7 pos (38.0 61.0 44.0 138.0 43.0 2.0 46.0 2.0 240.0)" | yarp rpc /ctpservice/left_arm/rpc  
-    echo "ctpq time 1.5 off 0 pos (-46.0 16.0 25.0 100.0 -44.0 0.0 0.0 38.0 61.0 44.0 138.0 43.0 2.0 46.0 2.0 240.0)" | yarp rpc /ctpservice/left_arm/rpc  
+    echo "ctpq time 0.2 off 7 pos (38.0 61.0 44.0 138.0 43.0 2.0 46.0 2.0 240.0)" | yarp rpc /ctpservice/left_arm/rpc
+    echo "ctpq time 1.5 off 0 pos (-46.0 16.0 25.0 100.0 -44.0 0.0 0.0 38.0 61.0 44.0 138.0 43.0 2.0 46.0 2.0 240.0)" | yarp rpc /ctpservice/left_arm/rpc
     sleep 1.5
     echo "ctpq time 0.7 off 0 pos (-13.0 -1.0 0.0 19.0 0.0 9.0)" | yarp rpc /ctpservice/head/rpc
     sleep 1.5
     echo "set track 1" | yarp rpc /iKinGazeCtrl/rpc
     echo "set all ang" | yarp rpc /icub/face/emotions/in
-    echo "ctpq time 1.0 off 0 pos (-77.0 16.0 10.0 47.0 82.0 -23.0 7.0 38.0 61.0 44.0 138.0 43.0 2.0 46.0 2.0 240.0)" | yarp rpc /ctpservice/left_arm/rpc  
+    echo "ctpq time 1.0 off 0 pos (-77.0 16.0 10.0 47.0 82.0 -23.0 7.0 38.0 61.0 44.0 138.0 43.0 2.0 46.0 2.0 240.0)" | yarp rpc /ctpservice/left_arm/rpc
     sleep 3.0
     echo "set all hap" | yarp rpc /icub/face/emotions/in
 }
@@ -96,18 +96,18 @@ look_you() {
     echo "set mou neu" | yarp rpc /icub/face/emotions/in
     echo "set reb neu" | yarp rpc /icub/face/emotions/in
     echo "set leb neu" | yarp rpc /icub/face/emotions/in
-    echo "ctpq time 0.2 off 7 pos (38.0 61.0 44.0 138.0 43.0 2.0 46.0 2.0 240.0)" | yarp rpc /ctpservice/left_arm/rpc  
-    echo "ctpq time 1.0 off 0 pos (-24.0 24.0 -9.0 106.0 -43.0 16.0 -10.0 38.0 61.0 44.0 138.0 43.0 2.0 46.0 2.0 240.0)" | yarp rpc /ctpservice/left_arm/rpc  
+    echo "ctpq time 0.2 off 7 pos (38.0 61.0 44.0 138.0 43.0 2.0 46.0 2.0 240.0)" | yarp rpc /ctpservice/left_arm/rpc
+    echo "ctpq time 1.0 off 0 pos (-24.0 24.0 -9.0 106.0 -43.0 16.0 -10.0 38.0 61.0 44.0 138.0 43.0 2.0 46.0 2.0 240.0)" | yarp rpc /ctpservice/left_arm/rpc
     sleep 1.0
     look_at -0.15 -0.7 0.12
     sleep 0.5
     #echo "set track 1" | yarp rpc /iKinGazeCtrl/rpc
     echo "set all ang" | yarp rpc /icub/face/emotions/in
-    echo "ctpq time 1.0 off 0 pos (-61.0 37.0 -16.0 52.0 87.0 -28.0 11.0 38.0 61.0 44.0 138.0 43.0 2.0 46.0 2.0 240.0)" | yarp rpc /ctpservice/left_arm/rpc  
+    echo "ctpq time 1.0 off 0 pos (-61.0 37.0 -16.0 52.0 87.0 -28.0 11.0 38.0 61.0 44.0 138.0 43.0 2.0 46.0 2.0 240.0)" | yarp rpc /ctpservice/left_arm/rpc
     sleep 1.5
     echo "set all hap" | yarp rpc /icub/face/emotions/in
     sleep 1.0
-    echo "ctpq time 1.5 off 0 pos (0.0 34.0 4.0 69.0 9.0 0.0 -1.0 54.0 30.0 9.0 49.0 33.0 49.0 30.0 26.0 73.0)" | yarp rpc /ctpservice/left_arm/rpc  
+    echo "ctpq time 1.5 off 0 pos (0.0 34.0 4.0 69.0 9.0 0.0 -1.0 54.0 30.0 9.0 49.0 33.0 49.0 30.0 26.0 73.0)" | yarp rpc /ctpservice/left_arm/rpc
     sleep 1.6
 }
 
@@ -115,7 +115,7 @@ home_seated_left() {
     echo "set mou neu" | yarp rpc /icub/face/emotions/in
     echo "ctpq time 1.0 off 0 pos (-9.0 0.0 -36.0 3.0 -1.0 12.0)" | yarp rpc /ctpservice/head/rpc
     echo "ctpq time 1.5 off 0 pos (-37.0 28.0 33.0 51.0 1.0 5.0 25.0 30.0 29.0 8.0 11.0 3.0 0.0 0.0 0.0 0.0)" | yarp rpc /ctpservice/left_arm/rpc
-    sleep 2.0  
+    sleep 2.0
 }
 
 greet_left_thumb_up() {

--- a/app/scripts/shells/trento.sh
+++ b/app/scripts/shells/trento.sh
@@ -54,7 +54,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 
@@ -464,7 +464,7 @@ seq_01() {
 
 seq_02() {
     speak "Mi chiamo aicab!"
-    speak "Piacere di conoscervi!" 
+    speak "Piacere di conoscervi!"
     sleep 1.0 && blink && hello_both
     wait_till_quiet
     go_home_walking
@@ -525,7 +525,7 @@ seq_10() {
 
 #### Day 01/06
 seq2_01() {
-    speak "Buongiorno a tutti!" && hello_both 
+    speak "Buongiorno a tutti!" && hello_both
     go_home_walking
     speak "Io sono aicab, il robot dalle sembianze di un bambino."
     sleep 1.0 && blink

--- a/app/scripts/shells/udine_danieli.sh
+++ b/app/scripts/shells/udine_danieli.sh
@@ -53,7 +53,7 @@ speak() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 
@@ -357,8 +357,8 @@ show_agitation() {
     echo "bind roll 0.0 0.0" | yarp rpc /iKinGazeCtrl/rpc
     sleep 0.2
     echo "bind yaw 0.0 0.0" | yarp rpc /iKinGazeCtrl/rpc
-    sleep 0.2    
-    
+    sleep 0.2
+
     gaze "look -30.0 0.0 5.0"
     sleep 1.5
     gaze "look 30.0 0.0 5.0"
@@ -374,7 +374,7 @@ show_agitation() {
     sleep 0.2
     echo "clear yaw" | yarp rpc /iKinGazeCtrl/rpc
     sleep 0.2
-    
+
     echo "ctpq time 1.0 off 0 pos (0.0 0.0 0.0)" | yarp rpc /ctpservice/torso/rpc
 }
 
@@ -429,7 +429,7 @@ sequence_01() {
 
 sequence_02() {
     speak "Senti Corsini!"
-    echo "ctpq time 1.2 off 0 pos (30.0 0.0 0.0)" | yarp rpc /ctpservice/torso/rpc    
+    echo "ctpq time 1.2 off 0 pos (30.0 0.0 0.0)" | yarp rpc /ctpservice/torso/rpc
     echo "look ang (abs 60.0 0.0 3.0)" | yarp rpc /iKinGazeCtrl/rpc
     sleep 1.0
     wait_till_quiet
@@ -440,9 +440,9 @@ sequence_02() {
     wait_till_quiet
 
     speak "Andiamo a lavorare assieme nel gioint lab appena creato"
-    echo "ctpq time 1.0 off 0 pos (0.0 0.0 0.0)" | yarp rpc /ctpservice/torso/rpc        
+    echo "ctpq time 1.0 off 0 pos (0.0 0.0 0.0)" | yarp rpc /ctpservice/torso/rpc
     echo "look ang (abs 0.0 0.0 3.0)" | yarp rpc /iKinGazeCtrl/rpc
-    echo "ctpq time 1.5 off 0 pos (-48.1752 17.8473 47.082 82.1615 -34.2005 9.58559 25.5543 14.4965 40.0 70.0 3.0 10.0 15.0 20.0 25.0 50.0)" | yarp rpc /ctpservice/right_arm/rpc 
+    echo "ctpq time 1.5 off 0 pos (-48.1752 17.8473 47.082 82.1615 -34.2005 9.58559 25.5543 14.4965 40.0 70.0 3.0 10.0 15.0 20.0 25.0 50.0)" | yarp rpc /ctpservice/right_arm/rpc
     sleep 2.0
     go_home_helper 2.0
     wait_till_quiet
@@ -450,8 +450,8 @@ sequence_02() {
     speak "dopo tanti anni che mi dici che sono ancora troppo immaturo, non vedo l'ora di andare in impianto, e farti vedere di cosa sono capace."
     sad
     echo "look ang (abs 0.0 -30.0 3.0)" | yarp rpc /iKinGazeCtrl/rpc
-    echo "ctpq time 1.5 off 0 pos (-29.3665 24.0876 10.2558 101.937 -60.0459 25.1423 0.505372 30.4981 62.3201 0.0 5.09218 0.0 0.0 0.0 0.0 0.928347)" | yarp rpc /ctpservice/right_arm/rpc 
-    echo "ctpq time 1.5 off 0 pos (-29.3665 24.0876 10.2558 101.937 -60.0459 25.1423 0.505372 30.4981 62.3201 0.0 5.09218 0.0 0.0 0.0 0.0 0.928347)" | yarp rpc /ctpservice/left_arm/rpc 
+    echo "ctpq time 1.5 off 0 pos (-29.3665 24.0876 10.2558 101.937 -60.0459 25.1423 0.505372 30.4981 62.3201 0.0 5.09218 0.0 0.0 0.0 0.0 0.928347)" | yarp rpc /ctpservice/right_arm/rpc
+    echo "ctpq time 1.5 off 0 pos (-29.3665 24.0876 10.2558 101.937 -60.0459 25.1423 0.505372 30.4981 62.3201 0.0 5.09218 0.0 0.0 0.0 0.0 0.928347)" | yarp rpc /ctpservice/left_arm/rpc
     sleep 2.0
     echo "look ang (abs 0.0 0.0 3.0)" | yarp rpc /iKinGazeCtrl/rpc
     smile

--- a/modules/funnyThingsApp/script/funnythings.sh
+++ b/modules/funnyThingsApp/script/funnythings.sh
@@ -61,7 +61,7 @@ speak_wait() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 

--- a/modules/funnyThingsApp/script/funnythings_template.sh
+++ b/modules/funnyThingsApp/script/funnythings_template.sh
@@ -61,7 +61,7 @@ speak_wait() {
 }
 
 blink() {
-    echo "blink" | yarp rpc /iCubBlinker/rpc
+    echo "blink_single" | yarp rpc /iCubBlinker/rpc
     sleep 0.5
 }
 

--- a/modules/iCubBlinker/iCubBlinker.thrift
+++ b/modules/iCubBlinker/iCubBlinker.thrift
@@ -34,7 +34,7 @@ service iCubBlinker_IDL
   * Performs a single blink.
   * @return true/false on success/failure.
   */
-  bool blink();
+  bool blink_single();
 
   /**
   * Performs a fast blink.

--- a/modules/iCubBlinker/main.cpp
+++ b/modules/iCubBlinker/main.cpp
@@ -520,7 +520,7 @@ public:
         return res;
     }
 
-    bool blink()
+    bool blink_single()
     {
         t0=Time::now();
         return doBlink();

